src/test/java/teammates/test/cases/webapi/UpdateFeedbackResponseActionTest.java

```diff
@@ -158,7 +215,273 @@ public void testExecute_instructorFeedbackSubmissionMcqGenerateOptionsForTeams_s
     @Test
     @Override
     protected void testAccessControl() throws Exception {
-        // TODO
+
+        ______TS("wrong giver type");
+
+        loginAsStudent(student1InCourse1.getGoogleId());
+
+        String[] wrongGiverTypeParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+        };
+
+        verifyCannotAccess(wrongGiverTypeParams);
+
+        ______TS("preview mode, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+
+        String[] previewParams = new String[] {
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.toString(),
+                Const.ParamsNames.FEEDBACK_RESPONSE_ID, typicalResponse.getId(),
+                Const.ParamsNames.PREVIEWAS, instructor1OfCourse1.email,
+        };
+
+        verifyCannotAccess(previewParams);
+
+        ______TS("response in session not open, cannot access");
+
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
```

Comment:
Already logged in?

Reply:
Yea this would make more sense when I separate the test cases into their individual methods haha